src/server/replica.cc

```diff
@@ -852,10 +892,11 @@ void DflyShardReplica::ExecuteTxWithNoShardSync(TransactionData&& tx_data, Conte
 }
 
 bool DflyShardReplica::InsertTxToSharedMap(const TransactionData& tx_data) {
-  std::lock_guard lk{multi_shard_exe_->map_mu};
-
+  multi_shard_exe_->map_mu.lock();
   auto [it, was_insert] =
       multi_shard_exe_->tx_sync_execution.emplace(tx_data.txid, tx_data.shard_cnt);
+  multi_shard_exe_->map_mu.unlock();
```

Comment:
Why not `unlock()` previously-existed `lk`? Any reason not to RAII here?

Reply:
Just needed to switch into `unique_lock` to unlock the unlock method :)